imagepuller/internal/service/verify_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/burgerdev/evil-registry/registry"
+	"github.com/containers/storage"
+	"github.com/opencontainers/go-digest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type StubStore struct {
+	putLayerLayer digest.Digest
+}
+
+func (s *StubStore) AddNames(_ string, _ []string) error {
+	panic("not implemented")
+}
+
+func (s *StubStore) CreateImage(_ string, _ []string, _, _ string, _ *storage.ImageOptions) (*storage.Image, error) {
+	panic("not implemented")
+}
+
+func (s *StubStore) CreateContainer(_ string, _ []string, _, _, _ string, _ *storage.ContainerOptions) (*storage.Container, error) {
+	panic("not implemented")
+}
+
+func (s *StubStore) Lookup(_ string) (string, error) {
+	panic("not implemented")
+}
+
+func (s *StubStore) Mount(_, _ string) (string, error) {
+	panic("not implemented")
+}
+
+func (s *StubStore) PutLayer(_, _ string, _ []string, _ string, _ bool, _ *storage.LayerOptions, _ io.Reader) (*storage.Layer, int64, error) {
+	return &storage.Layer{CompressedDigest: s.putLayerLayer}, 0, nil
+}
+
+func (s *StubStore) RemoveNames(_ string, _ []string) error {
+	panic("not implemented")
+}
+
+func TestGetAndVerifyImage_EvilRegistry(t *testing.T) {
+	log := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+		Level: slog.LevelInfo,
+	}))
+
+	tests := map[string]struct {
+		digest  string
+		wantErr string
+	}{
+		"missing digest is rejected": {
+			digest:  "",
+			wantErr: "parsing image digest",
+		},
+		"wrong manifest digest is caught": {
+			digest:  "sha256:6ad6bbb5735b84b10af42d2441e8d686b1d9a6cbf096b53842711ef5ddabd28d",
```

Comment:
Consider adding a comment along the lines of "the evil registry responds to unknown digests with a default manifest".